src/org/labkey/test/tests/StudySurveyTest.java

```diff
@@ -0,0 +1,139 @@
+package org.labkey.test.tests;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.categories.DailyB;
+import org.labkey.test.pages.EditDatasetDefinitionPage;
+import org.labkey.test.util.DataRegionTable;
+import org.labkey.test.util.PortalHelper;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+
+@Category({DailyB.class})
+@BaseWebDriverTest.ClassTimeout(minutes = 10)
+public class StudySurveyTest extends BaseWebDriverTest
+{
+    PortalHelper portalHelper = new PortalHelper(this);
+    private String datasetName = "SampleDataset";
+    private String surveyDesignName = "Dataset Survey Design";
+
+    @BeforeClass
+    public static void doSetup()
+    {
+        StudySurveyTest initTest = (StudySurveyTest) getCurrentTest();
+        initTest.setupProject();
+    }
+
+    public static String getDate(int incrementDays)
+    {
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        LocalDateTime date = LocalDateTime.now().plusDays(incrementDays);
+        return dtf.format(date);
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return "Study Survey Test Project";
+    }
+
+    @Override
+    protected BrowserType bestBrowser()
+    {
+        return BrowserType.CHROME;
+    }
+
+    private void setupProject()
+    {
+        _containerHelper.createProject(getProjectName(), "Study");
+        clickButton("Create Study");
+        click(Locator.radioButtonById("continuousTimepointType"));
+        clickButton("Create Study");
+        _containerHelper.enableModule("Survey");
+
+        goToProjectHome();
+        portalHelper.addWebPart("Datasets");
+        portalHelper.addWebPart("Survey Designs");
```

Comment:
Each call to `addWebPart` will enter and exit page admin mode. If you explicitly enable and disable page admin mode you can save a handful of unnecessary navigations.
```suggestion
        portalHelper.enterAdminMode();
        portalHelper.addWebPart("Datasets");
        portalHelper.addWebPart("Survey Designs");
        portalHelper.exitAdminMode();
```

Reply:
Done